pkg/compactor/blocks_cleaner_test.go

```diff
@@ -100,6 +108,7 @@ func testBlocksCleanerWithOptions(t *testing.T, options testBlocksCleanerOptions
 		CleanupInterval:                    time.Minute,
 		CleanupConcurrency:                 options.concurrency,
 		BlockDeletionMarksMigrationEnabled: options.markersMigrationEnabled,
+		TenantCleanupDelay:                 0, // immediately
```

Comment:
Would be great to set this to an higher value (adjusting `user-4` `FinishedTime` accordingly), and add another user which has the `FinishedTime` < delay, to make sure it's not cleaned up.